### ISSUE TITLE
Search results in the repos input are confusing

### ISSUE THREAD

**Issue by macraig** [sourcegraph/sourcegraph#23442]
#### Feedback
Internal feedback submitted by @slimsag 
Slack thread reference: https://sourcegraph.slack.com/archives/C0W2E592M/p1626375636074200 

When I search in the repos input on the search page, I do not find the repos I am expecting to find. I know why this is, but as a user this is really confusing.

<img width="1081" alt="Screenshot 2021-08-02 at 10 58 47" src="https://user-images.githubusercontent.com/1657213/127834053-941aa86f-fe75-4258-a7de-e959d43a8d31.png">
<img width="578" alt="Screenshot 2021-08-02 at 10 59 04" src="https://user-images.githubusercontent.com/1657213/127834094-584529e1-0e15-4103-9a29-990cfff609f3.png">

I just want it to find the Sourcegraph repository or github org and let me search within that. I do not see a good reason for it to be limited to just repositories in my query at all.

Relates to https://github.com/sourcegraph/sourcegraph/issues/20158

**Comment by benvenker** [comment]
@slimsag in this case, would you want us to also carry over your search query to the target repo, or are you just wanting to use the filter as a shortcut to get to a new repo?

**Comment by slimsag** [comment]
I just expect it to e.g. add the relevant repo: filter to my search query
basically. Let me know if I’m not understanding the question correctly :)
-- 
Follow me on twitter @slimsag <https://twitter.com/slimsag>.

**Comment by benvenker** [comment]
Yeah I got you. I guess what I'm thinking technically here is that we'd then need to detect that we couldn't find a match in the contextual list, and then either bring back a list of repos that matched so you could pick one, or just provide a little link button that said "search the 'sourcegraph' repo" in this case. Does that make sense/sound like what you're looking for?

**Comment by slimsag** [comment]
> we'd then need to detect that we couldn't find a match in the contextual list, and then either bring back a list of repos that matched so you could pick one

this is what I’d expect. The list should just be all repos on Sourcegraph, with priority given to repos found on your search. From a technical POV this is as simple as executing a repos search the same way we do on the site admin page if we didn’t find a match in the contextual list FWIW, probably <100 lines of code

**Comment by benvenker** [comment]
Ok sounds good, thanks for the clarification. I'll get this prioritized.
